powertools-idempotency/src/main/java/software/amazon/lambda/powertools/idempotency/internal/IdempotencyHandler.java

```diff
@@ -127,6 +127,8 @@ private Object handleForStatus(DataRecord record) {
         Class<?> returnType = ((MethodSignature) pjp.getSignature()).getReturnType();
         try {
             LOG.debug("Response for key '{}' retrieved from idempotency store, skipping the function", record.getIdempotencyKey());
+            if (returnType.equals(String.class))
```

Comment:
This changed line is in the stack trace of the error I hit running `IdempotencyE2ET` locally.

Reply:
not this line, error is on readValue (line 132)